src/theory/quantifiers/ceg_instantiator.cpp

```diff
@@ -293,17 +294,25 @@ bool CegInstantiator::doAddInstantiation( SolvedForm& sf, unsigned i, unsigned e
             for (unsigned j = 0; j<ita->second.size(); j++) {
               Node lit = ita->second[j];
               if( std::find( lits.begin(), lits.end(), lit )==lits.end() ){
```

Comment:
I don't really like that `std::find` is used to check if a `lit` in vector `lits`. Does `lits` have to be a vector at all? Can you not use an std::unordered_set here - then the lookup would be constant time instead of a linear search?

Reply:
The order on lits is important, since alits and lits are parallel vectors. We could have another std::unordered_set here for the lookup, but then the data is doubled, I'm not sure which is better (this is a pretty low-level performance issue)